src/plugins/mer/meroptionswidget.cpp

```diff
@@ -249,60 +250,107 @@ void MerOptionsWidget::onSrcFolderApplyButtonClicked(const QString &newFolder)
     MerSdk *sdk = m_sdks[m_virtualMachine];
 
     if (newFolder == sdk->sharedSrcPath()) {
-        QMessageBox::information(this, tr("Choose a new folder"),
+        QMessageBox::information(this,
+                                 tr("Choose a new folder"),
                                  tr("The given folder (%1) is the current alternative source folder. "
                                     "Please choose another folder if you want to change it.").arg(sdk->sharedSrcPath()));
         return;
     }
 
-    if (!sdk->connection()->isVirtualMachineOff()) {
-        QPointer<QMessageBox> questionBox = new QMessageBox(QMessageBox::Question,
-                tr("Close Virtual Machine"),
-                tr("Close the \"%1\" virtual machine?").arg(m_virtualMachine),
-                QMessageBox::Yes | QMessageBox::No,
-                this);
-        questionBox->setInformativeText(
-                tr("Virtual machine must be closed before the source folder can be changed."));
-        if (questionBox->exec() != QMessageBox::Yes) {
-            // reset the path in the chooser
-            m_ui->sdkDetailsWidget->setSrcFolderChooserPath(sdk->sharedSrcPath());
-            return;
+    const bool vmIsOff = sdk->connection()->isVirtualMachineOff();
+    const bool vmWasClosed = !vmIsOff && showCloseVirtualMachineDialog(tr("Virtual machine must be closed before the source folder can be changed."));
```

Comment:
The `isVirtualMachineOff()` call is only used to find out if the "Virtual machine must be closed..." dialog should be shown to the user.

Another call is unnecessary, as `sdk->connection()->lockDown(true)` will return `true`if the virtual machine is already off, allowing the update to proceed, in case the VM was shut down manually while the dialog was on screen.

True, `vmWasClosed` is slightly misleading at the point of declaration. It gets this meaning on line 274, where it is used to determine if the corresponding "Start 'MerSDK' virtual machine?" dialog should be shown. I renamed the variable to `vmWillBeClosed` to be more clear that no VM shutdown has taken place yet when the variable is declared.